frontend/src/static/js/main.js

```diff
@@ -155,6 +156,37 @@ window.app = new window.Vue({
       }
     },
 
+    renderAuthorShipTab() {
+      const encodedInfo = window.location.hash.slice(1).split('&');
+      this.deserialize(encodedInfo);
+      const hash = window.hashParams;
+      if (hash.info) {
+        this.updateTabAuthorship(hash.info);
+      } else if (hash.isTabOpen === false) {
+        this.isTabActive = false;
+      }
+    },
+
+    deserialize(obj) {
+      const info = {};
+      obj.forEach((param) => {
+        const [key, val] = param.split('=');
+        if (key === 'info') {
```

Comment:
use flat structure instead, `tab-author` & `tab-repo` then put wrap it up into `updateTabAuthorship({author: tab-author, ...})`